Userland/Libraries/LibJS/Heap/Handle.h

```diff
@@ -60,6 +60,16 @@ class Handle {
     {
     }
 
+    Handle(GCPtr<T> cell)
+        : Handle(cell.ptr())
+    {
+    }
+
+    Handle(NonnullGCPtr<T> cell)
+        : Handle(*cell)
+    {
+    }
```

Comment:
I am confused in general why there are three ways to construct a `Handle` (the constructor, `Handle::create`, and `make_handle`), so I'm not sure if I should prefer one of the other ways or not.

Reply:
it's because of the way `HandleImpl` currently works, it pretends to store a possibly null Cell pointer but that doesn't actually work, so we have to add another layer that default-constructs the handle, keeping the `RefPtr<HandleImpl>` null. Not sure why @awesomekling did it like that initially, but I don't like it very much. It's a saved allocation for null handles I guess, but in reality these are rare.

I have a (stale) branch locally to clean that up, but it got annoying in a few places so I never finished that.

Reply:
Ah, it looks like the `Handle(T* cell)` ctor _also_ takes care of that. No idea why there's three functions for the same thing then...